### ISSUE TITLE
Make inegration server configurable

### ISSUE THREAD

**Issue by Bubu** [vector-im/riot-android#2657]
Afaik riot-android is currently always using New Vectors scalar, which is not open source. Consequently the app would need to get the [NonFreeNet](https://f-droid.org/wiki/page/Antifeature:NonFreeNet) Antifeature in F-Droid. 

This ticket is sparse on details but might be (part of) the solution to this: https://github.com/vector-im/riot-android/issues/2582

Any objections in tagging riot in F-Droid that way until this is resolved?

**Comment by ara4n** [comment]
sorry, missed this. yup, .well-known support would be a nice way to support easy configuration of alternative integ managers. we're going to be implementing .well-known support in the fairly near future (folks like the french government need it for autodiscovering servers in a standards-compliant manner), so it'll come along in the next month or two, i hope.

**Comment by Bubu** [comment]
I'll add the antifeature (and a note explaining why it's there and that the problem will be solved in the future) then for now.

**Comment by ara4n** [comment]
Yup, that seems plausible, although it also feels like a slightly strange distinction to make - for instance, the integration manager itself can connect to other non-free services (e.g. Google Docs); would that also make the overall app non-free?

Eitherway, we're working on making it configurable via .well-known URI at the least - the code for this just landed in PR on Riot/Web for instance, and should hit Riot/Android as soon as it's ported over.

**Comment by eighthave** [comment]
For a counter example, take K-9 Mail.  It can be used to access a GMail account, but it is also possible to run a 100% free software mail service.  And K-9 by default does not use any proprietary service.  So it is not marked with _NonFreeNet_, even though someone using it with only a GMail account would always be using proprietary software.

**Comment by hrj** [comment]
>  The Google Captcha shown by default when signing up via Riot is another such service.

Going on a slight tangent here, but is there interest in a free Captcha service within these circles?

We (myself and a couple of student interns) are in the early stages of developing [libreCaptcha](https://github.com/librecaptcha). It's a long way from being production ready, but if it sounds useful, please leave a :+1: here, or star the repo, etc.

**Comment by amiryal** [comment]
@Bubu I appreciate your effort to bring transparency. However, this message is confusing to anyone who doesn’t know what “the integration server” means or what its role is, which might lead to believe (as in my case before reading the LWN article) that the app is locked down to using only the preconfigured homeserver. Could you update the F-Droid anti-feature description to include a little bit more context and implications of this server being non-free, and how this “integration server” is separate and independent of the homeserver? Thanks!

**Comment by amiryal** [comment]
Actually, this seems to not be the issue any more. When ticking the box, in my newly installed app (version 0.8.21 from F-Droid), for “Use custom server options (advanced)”, I see an option to change the “Identity server” away from `https://vector.im`. Is this the “New Vector implementation” that was mentioned in the anti-feature description?

**Comment by mvgorcum** [comment]
I get that it is a but confusing. The identity server you mentioned is FOSS, and also not the same as the integration server. The identity server is there for contact discovery by 3pid (email or phone number) while the integration server provides widgets, such as jitsi, etherpad or stickers.

**Comment by LaysDragon** [comment]
Here is another opensource integration server.
https://github.com/turt2live/matrix-dimension/

**Comment by Bubu** [comment]
Was this solved during the recent privacy sprint? (I'm currently only using RiotX, so I didn't get around to checking.)

**Comment by Bubu** [comment]
This is still an open question for me (and consequently the F-Droid issue about the antifeature is still open https://gitlab.com/fdroid/fdroiddata/-/issues/1598).

I think it comes down to: Can I use Riot (and probably soon RiotX) with my own homeserver without connecting to the nonfree modular/scalar integration server.

I think this is the case now, but I'm not sure, I don't really use integrations.

**Comment by frankwalter1301** [comment]
This is very important

**Comment by goaaats** [comment]
Does this work now?

Element on the web discovers my custom integration server just fine, but Element on Android(Google Play, 1.1.0 [40101002]) is stuck on vector.

Seems as if it should be possible, seeing that it already asks you twice for privacy consent.
